src/graph/GraphService.cpp

```diff
@@ -10,14 +10,40 @@
 #include "graph/RequestContext.h"
 #include "graph/SimpleAuthenticator.h"
 #include "storage/client/StorageClient.h"
+#include "graph/GraphFlags.h"
+
+DECLARE_string(meta_server_addrs);
+DECLARE_bool(local_config);
 
 namespace nebula {
 namespace graph {
 
 Status GraphService::init(std::shared_ptr<folly::IOThreadPoolExecutor> ioExecutor) {
+    auto addrs = network::NetworkUtils::toHosts(FLAGS_meta_server_addrs);
+    if (!addrs.ok()) {
+        return addrs.status();
+    }
+
+    meta::MetaClientOptions options;
+    options.serviceName_ = "graph";
+    options.skipConfig_ = FLAGS_local_config;
+    metaClient_ = std::make_unique<meta::MetaClient>(ioExecutor,
+                                                     std::move(addrs.value()),
+                                                     options);
+    // load data try 3 time
+    bool loadDataOk = metaClient_->waitForMetadReady(3);
+    if (!loadDataOk) {
+        // Resort to retrying in the background
+        LOG(WARNING) << "Failed to synchronously wait for meta service ready";
+    }
+
     sessionManager_ = std::make_unique<SessionManager>();
-    executionEngine_ = std::make_unique<ExecutionEngine>();
     authenticator_ = std::make_unique<SimpleAuthenticator>();
+    pluginManager_ = std::make_unique<PluginManager>(metaClient_.get());
```

Comment:
Why not define pluginManager inside ExecutionEngine directly.

Reply:
Because the authentication module outside ExecutionEngine also needs pluginManager

Reply:
authentication module is also inside ExecutionEngine